components/eam/src/physics/cam/ndrop.F90

```diff
@@ -1038,152 +975,165 @@ subroutine dropmixnuc( &
 
-      end do ! old_cloud_nsubmix_loop
+      enddo ! old_cloud_nsubmix_loop
 
       ! evaporate particles again if no cloud
 
-      do k = top_lev, pver
-         if (cldn(i,k) == 0._r8) then
+      do kk = top_lev, pver
+         if (cldn(icol,kk) == 0._r8) then
             ! no cloud
-            qcld(k)=0._r8
+            qcld(kk)=0._r8
 
             ! convert activated aerosol to interstitial in decaying cloud
-            do m = 1, ntot_amode
-               mm = mam_idx(m,0)
-               raercol(k,mm,nnew)    = raercol(k,mm,nnew) + raercol_cw(k,mm,nnew)
-               raercol_cw(k,mm,nnew) = 0._r8
-
-               do l = 1, nspec_amode(m)
-                  mm = mam_idx(m,l)
-                  raercol(k,mm,nnew)    = raercol(k,mm,nnew) + raercol_cw(k,mm,nnew)
-                  raercol_cw(k,mm,nnew) = 0._r8
-               end do
-            end do
-         end if
-      end do
+            do imode = 1, ntot_amode
+               mm = mam_idx(imode,0)
+               raercol(kk,mm,nnew)    = raercol(kk,mm,nnew) + raercol_cw(kk,mm,nnew)
+               raercol_cw(kk,mm,nnew) = 0._r8
+
+               do lspec = 1, nspec_amode(imode)
+                  mm = mam_idx(imode,lspec)
+                  raercol(kk,mm,nnew)    = raercol(kk,mm,nnew) + raercol_cw(kk,mm,nnew)
+                  raercol_cw(kk,mm,nnew) = 0._r8
+               enddo
+            enddo
+         endif
+      enddo
 
       ! droplet number
 
-      ndropcol(i) = 0._r8
-      do k = top_lev, pver
-         ndropmix(i,k) = (qcld(k) - ncldwtr(i,k))*dtinv - nsource(i,k)
-         tendnd(i,k)   = (max(qcld(k), 1.e-6_r8) - ncldwtr(i,k))*dtinv
-         ndropcol(i)   = ndropcol(i) + ncldwtr(i,k)*pdel(i,k)
-      end do
-      ndropcol(i) = ndropcol(i)/gravit
+      ndropcol(icol) = 0._r8
+      do kk = top_lev, pver
+         ndropmix(icol,kk) = (qcld(kk) - ncldwtr(icol,kk))*dtinv - nsource(icol,kk)
+         tendnd(icol,kk)   = (max(qcld(kk), 1.e-6_r8) - ncldwtr(icol,kk))*dtinv
+         ndropcol(icol)   = ndropcol(icol) + ncldwtr(icol,kk)*pdel(icol,kk)
+      enddo
+      ndropcol(icol) = ndropcol(icol)/gravit
 
+!  NOTE FOR C++ PORT:  prog_modal_aero is always true in our tests.
       if (prog_modal_aero) then
 
          raertend = 0._r8
          qqcwtend = 0._r8
 
-         do m = 1, ntot_amode
-            do l = 0, nspec_amode(m)
+         do imode = 1, ntot_amode
+            do lspec = 0, nspec_amode(imode)
 
-               mm   = mam_idx(m,l)
-               lptr = mam_cnst_idx(m,l)
+               mm   = mam_idx(imode,lspec)
+               lptr = mam_cnst_idx(imode,lspec)
+               qqcwtend(top_lev:pver) = (raercol_cw(top_lev:pver,mm,nnew) - qqcw(mm)%fld(icol,top_lev:pver))*dtinv
 
-               raertend(top_lev:pver) = (raercol(top_lev:pver,mm,nnew) - raer(mm)%fld(i,top_lev:pver))*dtinv
-               qqcwtend(top_lev:pver) = (raercol_cw(top_lev:pver,mm,nnew) - qqcw(mm)%fld(i,top_lev:pver))*dtinv
+               if( lspec.eq.0 ) then
```

Comment:
I almost forgot this, but we are suggested to use == to replace .eq.

Reply:
Done.